homeassistant/components/miflora/sensor.py

```diff
@@ -82,7 +84,7 @@
 
     for parameter in config[CONF_MONITORED_CONDITIONS]:
         name = SENSOR_TYPES[parameter][0]
-        unit = SENSOR_TYPES[parameter][1]
+        unit = hass.config.units.temperature_unit if name == "Temperature" else SENSOR_TYPES[parameter][1]
```

Comment:
Please check the `parameter` instead of `name`. `parameter` should be `"temperature"`.

Reply:
sure - updating